### DISCUSSION TITLE
North node calculation

### DISCUSSION THREAD

**Discussion by gipsea** [skyfielders/python-skyfield#631] [Q&A]
Hi there,

I'm looking for feedback/leads on how I can calculate the Moon nosrth node longitude. In the almanac section there is the function that calculate this point only when Moon pass the ecliptic.

I'm looking for somenting that following by instance this [wiki page](https://en.wikipedia.org/wiki/Longitude_of_the_ascending_node) give me the possibility to calculate the node position at any given date&time

I'll be happy to receive guidance/hints or any lead which can help me to develop the function.

Thanks everybody

**Comment by voronY** [comment]
Hello, look at the function swe_nod_aps in Swiss Ephemeris which calculates mean and osculating Nodes [link removed]

**Reply by gipsea** [reply]
Hi @voronY 

I was just taking a look at this functions when you posted. I made a test script and I found these issues:

- Julian time: there is the difference of about 1.5 (days i guess) between skyfield and swisseph.
- time in skyfield is time class and not just real (float) number
- swisseph do not accept a list of time used by find_XXXX function
This is the code I used and where changing few things you can find the above error/issues

```
import datetime
from skyfield.api import load
from skyfield.searchlib import find_discrete, find_minima, find_maxima

import swisseph as swe
swe.set_ephe_path('/usr/share/ephe')


ts = load.timescale()


#variable
data_inizio = datetime.date(year=2021,month=8, day=1)
data_fine = datetime.date(year=2021,month=8,day=31)

t1 = ts.utc(data_inizio)
t2 = ts.utc(data_fine)

true_node_sid={}
mean_node_long={}    
# l_t: test if swe.calc_ut accept list insted single time. below generate the date list
l_t=[]
data = data_inizio
while data <= data_fine:
    data_year= int(data.strftime('%Y'))
    data_month= int(data.strftime('%m'))
    data_day = int(data.strftime('%d'))
    l_t.append(swe.julday(data_year, data_month, data_day))
    data = data+datetime.timedelta(days=1)
# l_long_m_node = swe.calc_ut(l_t,10) uncommenting this line will return error about using list instead real float number
print("Longitudine del north node")
data = data_inizio
while data <= data_fine:
    data_year= int(data.strftime('%Y'))
    data_month= int(data.strftime('%m'))
    data_day = int(data.strftime('%d'))
    t = swe.julday(data_year, data_month, data_day)
    t_skyf=ts.utc(data)
    long_m_node = swe.calc_ut(t,10)
    print(t,type(t))
    print(t_skyf,type(t_skyf))
    long_m_node = swe.calc_ut(t_skyf,10)
    long = long_m_node[0][0]
    data = data+datetime.timedelta(days=1)
    
swe.close()
```
questions:

- time: is there any chance to convert skyfild julian time type into float? I tried few tricks and possibly my python knowledge is limited  and run out of options. If I can solve at least this issue I can arrange a work around for the find_XXXX function (done already on different own functions)
- julian day difference: I've no clue why there is such difference. has anyone a clue? swisseph use Ephemeridis time and TT. again not sure what the difference but I'm interested to understand the implication. If I manage to get time as float from skyfield I still have to possibly 'adjust' it before using on swisseph
- data: is the below code the best way to get year, month and day from a data?
 ```
data_year= int(data.strftime('%Y'))
data_month= int(data.strftime('%m'))
data_day = int(data.strftime('%d'))
```
if there will be no way to use jultime from skyfiled I'll need to convert this value back to data, extract Y,M,D and convert in jul time in swisseph to find the infromation I'm looking for. A best tuned code will reduce CPU use/time

sorry for such lenghty reply and I hope that with the script everything make a little sense

cheers

I know there are several issues but all can bereferred to the possibility to use julian date from skyfield with swisseph

**Comment by gipsea** [comment]
little update about Julian time difference.
I realized SWiss eph uses noon time 1200 and skyfiled use midnight. the actual difference is .5

Still working on how to get the julian date in skyfild as float number rather a time class

cheers

**Comment by gipsea** [comment]
One more brick in the wall.

reading more carefully the time doc page with specific attention to the different output possible I find the tt attribute in class time which return numpy.float value.

I tweaked the above script and it works so far.

At the moment I consider all my question answered.

Cheers
